### DISCUSSION TITLE
Wishlist: Direct communication with Quarto from R comments

### DISCUSSION THREAD

**Discussion by harrelfe** [quarto-dev/quarto-cli#3094] [Feature Requests]
I'd like to be able to do things such as the following, where #| is used to start markdown text.  This assumes the user doesn't have any "regular" R comments that start with #| and it also assumes that any lines beginning with #| after the first non-yaml line in a chunk are treated as special.

When the markdown specifies a marginal note, I'd like that note to vertically align with the code to which it applies in the code listing.  A global option might be implemented to tell Quarto to not also print the markdown in the code listing when it also generates markdown output.

````
```{r}
x <- runif(100)
y <- runif(100)
range(x)  #|[`range` is in package `base`]{.aside}
summary(data.frame(x,y)) #|^[footnote text ...]
lm(y ~ x)  #|[`lm` is in package `stats`]{.aside}
# In the following, multiple lines are passed to Quarto
# We'd assume that #| appearing after any non-yaml line will be special
#| * bullet list line 1
#| * bullet list line 2
# The multiple lines could also contain divs for special treatment such as `.column-margin`
```
````

**Comment by mcanouil** [comment]
Hum interesting feature request, it's really close to what `knitr::spin` does using an R script and roxygen2 style comment.
Maybe simpler/better to actually translate `knitr::spin` to Quarto.

One issue with this feature is: can it work with Jupyter? I am not so sure ...
Also, it kind of mixed up comments, markup text and code which might be very confusing to user in addition to make it hard to parse properly.

**Reply by harrelfe** [reply]
I see where it could be confusing but the `.aside` approach would be very helpful.   Another approach that may work and would address the issues you raised is to create a `Quarto` api whereby calling a certain function (with character string arguments) within a chunk would cause `Quarto/knitr` to  render the markup strings as soon as possible.

**Reply by mcanouil** [reply]
The issue, is that you want code to behave as if `output: true` but at the same time `output: asis`.
With `output: asis` you can run code, display/render HTML (or LaTeX), but the default print of objects will be messed up.


Why not write the following? I find it way more clear what's code and what's markup text.

````
```{r}
x <- runif(100)
y <- runif(100)
range(x)
```

[`range` is in package `base`]{.aside}

```{r}
summary(data.frame(x,y))
```

```{r}
lm(y ~ x)
```

[`lm` is in package `stats`]{.aside}
````

To note, aside in spans and divs have currently a different behaviour (#2934).

The confusing argument is not the main concern I think. Quarto is language agnostic. This feature is not. Maybe it is a `knitr` feature request instead? 🤔

**Reply by harrelfe** [reply]
It night be logical as a `knitr` request.  Thanks for your thoughts.  I'm trying not to break up a logical chunk into a lot of chunks.

**Reply by harrelfe** [reply]
P.S.  In my old days of using $\LaTeX$ with `.Rnw` files `knitr` provided a way to run $\LaTeX$ from within R for the purpose of cross-referencing, e.g.

```
plot(x)    # Figure (*\ref{plot-label}*)
```

where the figure number appeared when the R comment was typeset.

**Reply by mcanouil** [reply]
😮 I learned something!
I did not use a lot Sweave, I wonder if this is still in `knitr`, maybe hidden somewhere.

**Reply by harrelfe** [reply]
Yes that works with `knitr`.
